neqo-http3/src/control_stream_local.rs

```diff
@@ -51,6 +63,45 @@ impl ControlStreamLocal {
                     self.buf = b;
                 }
             }
+            // only send priority updates if all buffer data has been sent
+            if self.buf.is_empty() {
+                self.send_priority_update(stream_id, conn, recv_conn)?
+            }
+        }
+        Ok(())
+    }
+
+    fn send_priority_update(
+        &mut self,
+        stream_id: u64,
+        conn: &mut Connection,
+        recv_conn: &mut HashMap<u64, Box<dyn RecvStream>>,
+    ) -> Res<()> {
+        // send all necessary priority updates
+        while let Some(update_id) = self.outstanding_priority_update.pop_front() {
+            let update_stream = match recv_conn.get_mut(&update_id) {
+                Some(update_stream) => update_stream,
+                None => continue,
+            };
+
+            // can assert and unwrap here, because priority updates can only be added to
+            // HttpStreams in [Http3Connection::queue_update_priority}
+            debug_assert!(matches!(
+                update_stream.stream_type(),
+                Http3StreamType::Http | Http3StreamType::Push
+            ));
+            let priority_handler = update_stream.http_stream().unwrap().priority_handler_mut();
+
+            // in case multiple priority_updates were issued, ignore now irrelevant
+            if let Some(hframe) = priority_handler.maybe_encode_frame(update_id) {
+                let mut enc = Encoder::new();
+                hframe.encode(&mut enc);
+                if conn.stream_send_atomic(stream_id, &enc)? {
+                    priority_handler.priority_update_sent();
+                } else {
+                    self.outstanding_priority_update.push_front(update_id);
```

Comment:
You need to exit loop here, otherwise this will loop for ever.

Please add a test for this. (you can set a small max_data limit by setting the parameter in ConnectionParameters: `ConnectionParameters::default().max_data(...)` and you can fill the flow control by sending stream data.